UndertaleModTool/MainWindow.xaml.cs

```diff
@@ -3093,135 +3095,156 @@ public async void UpdateApp(SettingsWindow window)
+                            // Read header
+                            response.EnsureSuccessStatusCode();
+                            long totalBytes = response.Content.Headers.ContentLength ?? throw new Exception("Missing content length");
+
+                            // Start reading content
+                            using Stream contentStream = await response.Content.ReadAsStreamAsync();
+                            const int downloadBufferSize = 8192;
+                            byte[] downloadBuffer = new byte[downloadBufferSize];
+
+                            // Download content and save to file
+                            using FileStream fs = new(downloadOutput, FileMode.Create, FileAccess.Write, FileShare.None, downloadBufferSize, true);
+                            int bytesRead = await contentStream.ReadAsync(downloadBuffer);
+                            long totalBytesDownloaded = 0;
+                            long bytesToUpdateProgress = totalBytes / 500;
+                            long bytesToProgressCounter = 0;
+                            while (bytesRead > 0)
                             {
-                                errMsg = "Failed to download new version of UndertaleModTool.\n" +
-                                         "Error - The SSL connection could not be established.";
-
-                                bool isWin7 = Environment.OSVersion.Version.Major == 6;
-                                string win7upd = "\nProbably, you need to install Windows update KB2992611.\n" +
-                                                 "Open the update download page?";
+                                // Write current data to file
+                                await fs.WriteAsync(downloadBuffer.AsMemory(0, bytesRead));
 
-                                if (isWin7)
+                                // Update progress
+                                totalBytesDownloaded += bytesRead;
+                                bytesToProgressCounter += bytesRead;
+                                if (bytesToProgressCounter >= bytesToUpdateProgress)
                                 {
-                                    if (this.ShowQuestion(errMsg + win7upd, MessageBoxImage.Error) == MessageBoxResult.Yes)
-                                        OpenBrowser("https://www.microsoft.com/en-us/download/details.aspx?id=44622");
-
-                                    window.UpdateButtonEnabled = true;
-                                    return;
+                                    bytesToProgressCounter -= bytesToUpdateProgress;
+                                    try
+                                    {
+                                        UpdateProgressStatus($"Downloaded MB: {(totalBytesDownloaded / bytesToMB).ToString("F2", CultureInfo.InvariantCulture)}");
+                                    }
+                                    catch { }
```

Comment:
?

Reply:
Honestly this was here in the code before and I'm afraid to remove it now. Guessing there's some rare chance WPF throws something here, but not sure what that is. Do you think I should remove it, though?

Reply:
Yeah id remove it. By looking at it, i cant ses any case where it would throw
And if it does, then thatll be useful for figuring it out when we dont silently catch!

Reply:
Alright, went ahead and removed it.